### PR TITLE
Set correct buffer size in into_options() on libcoap<4.3.5rc2

### PR DESCRIPTION
**Pull request by pulsastrix**
Versions of libcoap older than 4.3.5rc2 used a user-provided buffer for temporarily storing the options during conversion.

This buffer, however, must be larger than what the [libcoap documentation originally suggested](https://libcoap.net/doc/reference/4.3.4/group__uri__parse.html#gab12de398f21df10012700359bfdd92ad), as this buffer area must also hold the option headers.
If this is not done, trailing elements of the URI are silently removed.

This PR increases the used buffer size for this temporary buffer to a value that should be large enough to store all options that could be created from this URI.

Additionally, it adds a check for newer libcoap versions that don't use this buffer anymore.
In that case, we just pass an empty buffer instead, saving memory and avoiding some iterations over the URI path and query string to determine the buffer size.

### CODE REVIEW

File: libcoap/src/message/request.rs [hunk 1]
```diff
@@ -9,6 +9,9 @@
 
 use std::str::FromStr;
 
+use crate::error::OptionValueError;
+use crate::message::{construct_path_string, construct_query_string};
+use crate::session::CoapSessionCommon;
```
Comment (JKRhb) [inline libcoap/src/message/request.rs:14]: Was the reordering here intentional?

File: libcoap/src/message/request.rs [hunk 2]
```diff
@@ -428,7 +428,7 @@ impl CoapRequest {
     /// Converts this request into a [CoapMessage] that can be sent over a [CoapSession](crate::session::CoapSession).
     pub fn into_message(mut self) -> CoapMessage {
         if self.uri.is_proxy() {
-            self.pdu.add_option(CoapOption::ProxyUri(
+            self.pdu.add_option(CoapOption::ProxyScheme(
```
Comment (JKRhb) [inline libcoap/src/message/request.rs:431]: Hmm, is this change related to the main fix of this PR? Otherwise, it could also become its own commit, I guess.

File: libcoap/src/types.rs [hunk 3]
```diff
@@ -9,37 +9,37 @@
...
     os::raw::c_int,
     str::FromStr,
 };
-use std::ffi::CString;
-use std::fmt::{Display, Formatter};
-use std::marker::PhantomPinned;
-use std::ops::{Deref, DerefMut};
-use std::pin::Pin;
 
-use libc::{AF_INET, AF_INET6, c_ushort, in6_addr, in_addr, sa_family_t, sockaddr_in, sockaddr_in6, socklen_t};
```
Comment (JKRhb) [inline libcoap/src/types.rs:25]: Some unrelated changes here – maybe the reformatting could also be moved into its own commit?

File: libcoap/src/types.rs [hunk 4]
```diff
@@ -322,12 +322,14 @@ impl From<CoapProtocol> for CoapUriScheme {
 /// # Note on URI Length Limits
 ///
 /// Due to [the specified limits](https://datatracker.ietf.org/doc/html/rfc7252#section-5.10)
-/// of CoAP option lengths, URI path components for a request/location URI must not exceed 255
-/// bytes each, i.e. a full path with more than 255 bytes is fine, but each individual path segment
-/// must be smaller than 255 bytes.
+/// of CoAP option lengths, URI path components, query components and hostnames for a URI must not
```
Comment (JKRhb) [inline libcoap/src/types.rs:325]: Oxford comma and a small clarification.

```suggestion
/// of CoAP option lengths, the URI path components, query components, and hostnames for a URI must not
```

**Review by JKRhb — APPROVED** [review]
Great work finding and solving the issue here :) See a couple of small suggestions below, other than that: Looks good to me :)

File: libcoap/src/types.rs (hunk 3 above)
Comment (pulsastrix) [reply libcoap/src/types.rs:25]: I didn't make those changes intentionally, my IDE just randomly decided to "optimize" these imports.

I'll revert these changes, as they don't serve any real purpose.
